front/src/modules/trainschedule/components/ManageTrainSchedule/Itinerary/Itinerary.tsx

```diff
@@ -64,9 +77,8 @@ function Itinerary() {
 
   const inverseOD = () => {
     if (origin && destination) {
-      const newOrigin = { ...origin };
       dispatch(updateOrigin(destination));
-      dispatch(updateDestination(newOrigin));
+      dispatch(updateDestination(origin));
```

Comment:
You can't be sure that `origin` will not change, you have to use a third party variable to exchange values, here !
It'll be safer to keep old code.